src/Tracy/Dumper/Dumper.php

```diff
@@ -517,7 +518,7 @@ public static function encodeString(string $s, int $maxLength = null): string
 				$i = $len = 0;
 				$maxI = $maxLength * 4; // max UTF-8 length
 				do {
-					if (($s[$i] < "\x80" || $s[$i] >= "\xC0") && (++$len > $maxLength) || $i >= $maxI) {
+					if (($s[$i] < "\x80" || $s[$i] >= "\xC0") && ((++$len > $maxLength) || $i >= $maxI)) {
```

Comment:
why?

Reply:
Operations priority might differ from what programmer expect. Brackets are safe.

Reply:
if it's not related the phpstan error it should not be part of this PR